### PR TITLE
Added formatter to the ci run

### PR DESCRIPTION
**Pull request by BritikovKI**

### CODE REVIEW

**Review by aehyvari — COMMENTED** [review]
Maybe we can run the formatted completely independently, in parallel with the other checks?

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+            - image: cimg/base:stable
+              auth:
+                  username: mydockerhub-user
+                  password: $DOCKERHUB_PASSWORD
+              environment:
+                  CMAKE_BUILD_TYPE: Release
+                  FLAGS: -Wall -Wextra -Werror
+                  OSMT_INSTALL: ~/osmt-install
+                  USE_READLINE: OFF
+                  PARALLEL: OFF
```
Comment (aehyvari) [inline .circleci/config.yml:14]: I don't think these are needed?

File: .circleci/config.yml [hunk 2]
```diff
@@ -150,13 +175,14 @@ jobs:
...
         - build-recent:
-          filters:
-            <<: *filters-build-test
+              filters:
+                <<: *filters-build-test
         - build-macos:
-          filters:
-            <<: *filters-build-test
+              filters:
+                <<: *filters-build-test
```
Comment (aehyvari) [inline .circleci/config.yml:188]: Why these changes?

**Review by aehyvari — COMMENTED** [review]
Looks good, I left some comments.

File: .circleci/config.yml (hunk 1 above)
Comment (BritikovKI) [reply .circleci/config.yml]: Fixed

File: .circleci/config.yml (hunk 2 above)
Comment (BritikovKI) [reply .circleci/config.yml]: Fixed

**Review by aehyvari — APPROVED** [review]
LGTM!

**Comment by blishko** [comment]
@BritikovKI, can you delete the remote branches after merge? It keeps the view of the repository cleaner.

**Comment by BritikovKI** [comment]
> @BritikovKI, can you delete the remote branches after merge? It keeps the view of the repository cleaner.

Done, thanks for the reminder
